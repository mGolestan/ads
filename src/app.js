// @flow
import express from "express";
import bodyParser from "body-parser";
import compress from "compression";
import routes from "./routes";
import helmet from "helmet";
import cors from "cors";
import loadEnv from "./utils/loadEnv";
import { notFound, errResponse, logRequest } from "./middlewares";

const app = express();

app
  .use(
    helmet({
      dnsPrefetchControl: false,
      frameguard: false,
      noCache: true
    })
  )
  .use(compress())
  .use(cors("*"))
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({ extended: true }))
  .use(logRequest)
  .use("/", routes)
  .use(notFound)
  .use(errResponse);

module.exports = app;
