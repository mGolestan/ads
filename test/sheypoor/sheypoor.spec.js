// @flow

import request from "supertest";

import app from "../../src/app";

describe("GET /scrape/sheypoor", () => {
  it("returns a valid result on a valid request", () => {
    const city = "tehran";
    const q = "میز";
    return request(app)
      .get(`/scrape/sheypoor?city=${city}&q=${q}`)
      .expect(200)
      .then(res => {
        expect(res.body).toMatchSnapshot();
      });
  });

  it("returns BadRequest on an invalid city", () => {
    const city = "";
    const q = "میز";
    return request(app)
      .get(`/scrape/sheypoor?city=${city}&q=${q}`)
      .expect(400);
  });

  it("returns BadRequest on an invalid q parameter", () => {
    const city = "terhan";
    const q = "";
    return request(app)
      .get(`/scrape/sheypoor?city=${city}&q=${q}`)
      .expect(400);
  });

  it("returns BadRequest if not entered url querys", () => {
    return request(app)
      .get(`/scrape/sheypoor`)
      .expect(400);
  });
});
