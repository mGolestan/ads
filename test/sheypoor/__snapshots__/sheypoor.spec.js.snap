// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /scrape/sheypoor returns a valid result on a valid request 1`] = `
Object {
  "results": Array [],
  "version": 1,
}
`;
